### ISSUE TITLE
Improvements to distance functions?

### ISSUE THREAD

**Issue by karldw** [pysal/pysal#960]
#### More accurate distances:
The current implementation uses the approximation of a [spherical earth projected to a plane](https://en.wikipedia.org/wiki/Geographical_distance#Spherical_Earth_projected_to_a_plane), which "is very fast and produces [a] fairly accurate result for small distances."  Is it worth adding an option for more accurate calculations of longer distances? (Ref: https://github.com/geopy/geopy/pull/144.) Doing so might require shifting from a KD tree to, e.g., a [ball tree](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.BallTree.html#sklearn.neighbors.BallTree).

#### Faster, at current accuracy:
Since the square root doesn't change ordering, the arc length KDTree distance calculations could be sped up by skipping that step.  (For example, scikit-learn [takes this approach](https://github.com/scikit-learn/scikit-learn/blob/cb1b6c4734b2989ad0492b56c326858d030f3fa2/sklearn/neighbors/dist_metrics.pyx#L107).) Is that change worth the hassle?

**Comment by PranayMukherjee** [comment]
Interested to work on this issue. Could you elaborate the term "more accurate distances "?

**Comment by karldw** [comment]
Hi @PranayMukherjee!  I mean that accurately calculating distances on the Earth is hard.  It's computationally intensive to calculate the [distance between points on an ellipsoid](https://en.wikipedia.org/wiki/Geodesics_on_an_ellipsoid), so it's common to assume a sphere and work from there. Using a sphere works well most of the time; you would only bother with ellipsoidal distances if you need a great deal of accuracy. 

Other resources:
R's [lwgeom](https://github.com/r-spatial/lwgeom) package, which wraps liblwgeom.
PostGIS's distance functions: [sphere](https://postgis.net/docs/ST_DistanceSphere.html) vs. [spheroid](https://postgis.net/docs/ST_Distance_Spheroid.html).

**Comment by aghinsa** [comment]
Interested to work on this, is it still open?can i take it up?

**Comment by karldw** [comment]
Hi @aghinsa, I'm not sure. It looks like pysal is undergoing a bit of [refactoring](https://github.com/pysal/pysal/wiki/PEP-13:-Refactor-PySAL-Using-Submodules), so this question might be more appropriate for the [libpysal](https://github.com/pysal/libpysal) computational geometry submodule, `libpysal.cg`.
